medinify/sentiment/nn_review_classifier.py

```diff
@@ -0,0 +1,222 @@
+        text = ' '.join(filtered_words)
+
+        return text
+
+    def clean_text(self, text):
+        """ Takes a string, converts to lowercase, and removes all punctuation.
+
+        Args:
+            sent: String to be converted.
+            stop_words: Stop words to be removed. Defaults to None.
+        Returns:
+            The cleaned text.
+        """
+
+        translator = str.maketrans('', '', string.punctuation)
+        text = text.translate(translator).lower()
+
+        return text
+
+    def format_text(self, text, stop_words=None):
+        """ Takes a string, converts to lowercase, and removes all punctuation & stop words.
+
+        Args:
+            sent: String to be converted.
+            stop_words: Stop words to be removed. Defaults to None.
+        Returns:
+            The formatted text.
+        """
+
+
+        text = self.clean_text(text)
+
+        if stop_words is not None:
+            text = self.remove_stop_words(text, stop_words)
+
+        return ({word: True for word in nltk.word_tokenize(text)})
+
+    def parse_reviews(self, reviews_path):
+        """ Parses a CSV of reviews into a list of comments with rating.
+
+        Args:
+            reviews_file: Reviews file to be parsed.
+        Returns:
+            The list of comments.
+        """
+
+        reviews = []
+
+        with open(reviews_path, newline='') as reviews_file:
+            reader = csv.DictReader(reviews_file)
+
+            for row in reader:
+                reviews.append({'comment': row['comment'], 'rating': row['rating']})
+
+        return reviews
+    
+    def build_dataset(self, reviews, stop_words):
+         ## Parse and convert positive and negative examples
+        positive_comments = []
+        negative_comments = []
+
+        for review in reviews:
+            comment = review['comment']
+            rating = review['rating']
+
+            comment = self.format_text(comment, stop_words)
+
+            if float(rating) <= self.negative_threshold:
+                negative_comments.append((comment, 'neg'))
+            if float(rating) >= self.positive_threshold:
+                positive_comments.append((comment, 'pos'))
+
+        seed = 123
+        np.random.seed(seed)
+
+        print("Total Negative Instances:" + str(len(negative_comments)))
+        print("Total Positive Instances:" + str(len(positive_comments)))
+
+        negcutoff = math.floor(len(negative_comments) * 1)
+        poscutoff = math.floor(len(positive_comments) * 1)
+
+        neg_idx_train = sorted(random.sample(range(len(negative_comments)), negcutoff))
+        neg_train = [negative_comments[i] for i in neg_idx_train]
+
+        pos_idx_train = sorted(random.sample(range(len(positive_comments)), poscutoff))
+        pos_train = [positive_comments[i] for i in pos_idx_train]
+
+        dataset = neg_train + pos_train
+        return dataset
+
+    def nn_train(self, reviews_file):
```

Comment:
Should just call this train() in so it's more consistent with ReviewClassifier().